lib/unfurls/index.js

```diff
@@ -89,17 +97,73 @@ async function unfurlAction(req, res) {
     await storedUnfurl.destroy();
     return res.send({
       delete_original: true,
+      attachments: [new MutePromptsPrompt().getAttachment()],
     });
   }
+  if (res.locals.action === 'unfurl') {
+    const storedUnfurl = await Unfurl.findById(req.body.callback_id.replace('unfurl-', ''), {
+      include: [SlackWorkspace],
+    });
+    await storedUnfurl.deliver();
 
-  await storedUnfurl.deliver();
+    const { owner, repo } = githubUrl(storedUnfurl.url);
 
-  return res.send({
-    delete_original: true,
-  });
+    const { team } = await storedUnfurl.SlackWorkspace.client.team.info();
+
+    return res.send({
+      delete_original: true,
+      attachments: [new AutoUnfurlPrompt(owner, repo, storedUnfurl, team).getAttachment()],
+    });
+  }
+}
+
+async function unfurlAutoSetting(req, res) {
+  const [repoId, repoNameWithOwner] = req.body.callback_id.replace('unfurl-auto-', '').split('|');
+
+  if (res.locals.action === 'this-channel') {
+    await res.locals.slackUser.setAutomaticUnfurl(repoId, req.body.channel.id);
+    return res.send({
+      replace_original: true,
+      attachments: [new AutoUnfurlSettingConfirm(repoNameWithOwner, false).getAttachment()],
+    });
+  }
+  if (res.locals.action === 'all-channels') {
+    await res.locals.slackUser.setAutomaticUnfurl(repoId, 'all');
+    return res.send({
+      replace_original: true,
+      attachments: [new AutoUnfurlSettingConfirm(repoNameWithOwner, true).getAttachment()],
+    });
+  }
+}
+
+async function mutePrompts(req, res) {
+  if (res.locals.action === 'mute-24h') {
+    await res.locals.slackUser.update({
+      settings: {
+        muteUnfurlPromptsUntil: moment().add(24, 'hours').unix(),
+      },
+    });
+    return res.send({
+      replace_original: true,
+      attachments: [new MutePromptsSettingConfirm('for 24h').getAttachment()],
+    });
+  }
+  if (res.locals.action === 'mute-indefinitely') {
```

Comment:
There's quite a few conditionals here. Would it make sense to also try to route based on action?